src/commands/government/searchplate.js

```diff
@@ -20,15 +21,21 @@ module.exports = class searchPlate extends Command {
 
   async run ({ t, author, channel, message }, plt) {
     channel.startTyping()
-    const plate = await this.client.apis.sinesp.searchPlate(plt)
     try {
-      // vv  i know that's actually hardcoded :hahaa:
-      const carpic = await this.client.apis.gsearch.searchImage(`${plate.modelo} ${plate.anoModelo} a venda&imgSize=medium`)
+      const plate = await this.client.apis.consultaplaca.searchPlate(plt)
+      const mercosulPlate = await CanvasTemplates.plateMercosul(plate.placa, `https://cdn.jsdelivr.net/gh/bgeneto/bandeiras-br/imagens/${plate.uf.toUpperCase()}.png`)
+      const oldPlate = await CanvasTemplates.oldPlate(plate.placa.replace(/(....)$/, '-$1'), `${plate.uf} - ${plate.municipio.toUpperCase()}`)
+      const plateType = plate.extra.placa_nova !== 't' ? oldPlate : mercosulPlate
+      // TODO: find a better car image api
+      const carpic = await this.client.apis.gsearch.searchImage(`${plate.modelo} ${plate.anoModelo} webmotors`)
+      const attach = new MessageAttachment(plateType, 'plate.png')
       channel.send(
         new SwitchbladeEmbed()
-          .setDescription(`**${t('commands:searchplate.plate')}**: ${plate.placa}\n**${t('commands:searchplate.model')}**: ${plate.marca} ${plate.modelo}\n**${t('commands:searchplate.year')}**: ${plate.anoModelo}\n**${t('commands:searchplate.color')}**: ${plate.cor}\n**${t('commands:searchplate.city')}**: ${plate.municipio} - ${plate.uf}\n**${t('commands:searchplate.status')}**: ${(plate.situacao !== 'Roubo/Furto') ? t('commands:searchplate.notstolen') : `**${t('commands:searchplate.stolen')}**`}\n**${t('commands:searchplate.chassis')}**: ${cleanContent(plate.chassi, message)}\n\n**${t('commands:searchplate.image')}**:`)
+          .setDescription(`**${t('commands:searchplate.plate')}**: ${plate.placa}\n**${t('commands:searchplate.model')}**: ${plate.extra.marca_modelo.modelo}\n**${t('commands:searchplate.year')}**: ${plate.anoModelo}\n**${t('commands:searchplate.color')}**: ${plate.cor}\n**${t('commands:searchplate.type')}**: ${plate.extra.tipo_veiculo.tipo_veiculo} (${plate.extra.quantidade_passageiro} passengers)\n**${t('commands:searchplate.maxtorque')}**: ${plate.extra.potencia}bhp\n**${t('commands:searchplate.engineCapacity')}**: ${plate.extra.cilindradas}cc\n**${t('commands:searchplate.nationality')}**: ${plate.extra.nacionalidade.nacionalidade}\n**${t('commands:searchplate.fuelType')}**: ${plate.extra.combustivel.combustivel}\n**${t('commands:searchplate.city')}**: ${plate.municipio} - ${plate.uf}\n**${t('commands:searchplate.status')}**: ${(plate.situacao !== 'Roubo/Furto') ? t('commands:searchplate.notstolen') : `**${t('commands:searchplate.stolen')}**`}\n**${t('commands:searchplate.chassis')}**: ${cleanContent(plate.extra.chassi, message)}\n\n**${t('commands:searchplate.image')}**:`)
           .setImage(carpic.items[0].link || 'https://i.imgur.com/awMCwj3.jpg')
```

Comment:
Shouldn't we check if carpic.items exists before assuming it has a link property?

Reply:
that's why we have the ||

Reply:
and the google api handles it

Reply:
> that's why we have the ||

The || would only handle if items[0].link is a falsy value, but you're still trying to grab .link from items. if items were to be null then this would error, unless items cannot ever be null or not have the link property then we should check like `(carpic.items[0] ? carpic.items[0].link : undefined) || 'https://i.imgur.com/awMCwj3.jpg'`

Reply:
that's actually good, if you can make a suggestion, taria agradecido